pkg/drupal/dbadmincheck.go

```diff
@@ -0,0 +1,139 @@
+package drupal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os/exec"
+	"strings"
+
+	"github.com/salsadigitalauorg/shipshape/pkg/shipshape"
+	"github.com/salsadigitalauorg/shipshape/pkg/utils"
+)
+
+const AdminUser shipshape.CheckType = "drupal-admin-user"
+
+// AdminUserCheck fetches all role configurations from the database and verifies
+// they do not have is_admin set to true.
+type AdminUserCheck struct {
+	shipshape.CheckBase `yaml:",inline"`
+	DrushCommand        `yaml:",inline"`
+	// List of role names allowed to have is_admin set to true.
+	AllowedRoles []string `yaml:"allowed-roles"`
+	userRoles    []string
+	roleConfigs  map[string]bool
+}
+
+type roleConf struct {
+	IsAdmin bool `json:"is_admin"`
+	Name string `json:"id"`
+}
+
+// Init implementation for the drush-based user role config check.
+func (c *AdminUserCheck) Init(ct shipshape.CheckType) {
+	c.CheckBase.Init(ct)
+	c.RequiresDb = true
+}
+
+// Merge implementation for AdminUserCheck check.
+func (c *AdminUserCheck) Merge(mergeCheck shipshape.Check) error {
+	adminUserMergeCheck := mergeCheck.(*AdminUserCheck)
+	if err := c.CheckBase.Merge(&adminUserMergeCheck.CheckBase); err != nil {
+		return err
+	}
+
+	c.DrushCommand.Merge(adminUserMergeCheck.DrushCommand)
+	utils.MergeStringSlice(&c.AllowedRoles, adminUserMergeCheck.AllowedRoles)
+	return nil
+}
+
+// getActiveRoles runs the drush command to populate data for the roles config check.
+func (c *AdminUserCheck) getActiveRoles() map[string]string {
+  var err error
```

Comment:
The file does not seem to be formatted consistently in a number of places (at least in Github) - VSCode has the plugins to autoformat, but I think we'll have to add a step in the pipeline to also do that.

Reply:
Thanks for the review! Yes, I noticed GH shows it with a dodgy formatting, in my PhpStorm it looks formatted OK.
I will double-check in VSCode.

And yes, agreed about auto-formatting, thank would be neat!